crates/rpc/src/eth/account.rs

```diff
@@ -0,0 +1,42 @@
+use ethereum_rust_storage::Store;
+use serde_json::Value;
+use tracing::info;
+
+use crate::utils::RpcErr;
+use ethereum_rust_core::Address;
+
+use super::block::BlockIdentifier;
+
+pub struct GetBalanceRequest {
+    pub address: Address,
+    pub block: BlockIdentifier,
+}
+
+impl GetBalanceRequest {
+    pub fn parse(params: &Option<Vec<Value>>) -> Option<GetBalanceRequest> {
```

Comment:
any reason why this returns `Option` instead of `Result`? Not against it, just curious

Reply:
All failure cases lead to a `RpcError::BadParams` error, by returning an Option we can omit all the repeated error mapping and just map once where the function is called. We could also use an internal function and make `parse` return the `RpcError::BadParams`